pandas/tests/io/test_parquet.py

```diff
@@ -1131,6 +1131,33 @@ def test_infer_string_large_string_type(self, tmp_path, pa):
     #     assert result["strings"].dtype == "string"
     # FIXME: don't leave commented-out
 
+    def test_non_nanosecond_timestamps(self, tmp_path, pa):
+        # GH#49236
+        #
+        # pandas 1.x didn't support non-nanosecond datetimes.
+        # pyarrow.Table.to_pandas supports timestamp_as_object param to solve that issue
+        # https://arrow.apache.org/docs/python/generated/pyarrow.Table.html#pyarrow.Table.to_pandas
+        #
+        # This test tests that the current version of pandas
+        # supports non-nanosecond (microsecond in this case) datetimes,
+        # the code example from GH#49236 doesn't fail anymore,
+        # and timestamp_as_object param is not needed.
+        import pyarrow as pa
+        import pyarrow.parquet as pq
+
+        path = tmp_path / "non_nanosecond_timestamp.p"
```

Comment:
pandas already has a convenience context manager for creating temporary files - check out `tm.ensure_clean` from other tests

Reply:
Updated. Thanks!